Src/IronPython/Compiler/Parser.cs

```diff
@@ -1939,29 +1939,9 @@ public class Parser : IDisposable { // TODO: remove IDisposable
                 case TokenKind.Name:            // identifier
                     NextToken();
                     string name = (string)t.Value;
-                    ParserSink?.StartName(GetSourceSpan(), name);
-                    var fixedName = FixName(name);
-                    if (ShouldReplaceClassKeyword(fixedName)) {
-                        // if `__class__` variable is used in a class method, replace with `name_of_the_class`.
-                        ret = new NameExpression(CurrentClass.Name);
-                    } else {
-                        ret = new NameExpression(fixedName);
-                    }
+                    _sink?.StartName(GetSourceSpan(), name);
```

Comment:
This should be `ParserSink`
```suggestion
                    ParserSink?.StartName(GetSourceSpan(), name);
```